src/ray/protobuf/experimental/autoscaler.proto

```diff
@@ -17,7 +17,101 @@ option cc_enable_arenas = true;
 
 package ray.rpc.autoscaler;
 
-import "src/ray/protobuf/experimental/instance_manager.proto";
+// ============= Instance Definition ====================
+enum RayNodeKind {
+  UNKNOWN = 0;
+  HEAD = 1;
+  WORKER = 2;
+}
+// A node type that's available for the cluster.
+message InstanceType {
+  // the name of the instance type. e.g. "m4.large"
+  string type_name = 1;
+  RayNodeKind node_kind = 2;
+  // avaialble resources on the node. such as {"CPU": 4, "GPU": 1}
+  map<string, double> resources = 3;
+}
+
+// Represents the state of a launched instance.
+// An instance is considered launched as long as
+// it has a unique instance_id associated with it.
+//
+// Note a launched instance may be DEAD. In this case,
+// the state will be garbage collected after some timeout
+// period (by default 30 minutes).
+message Instance {
+  enum InstanceStatus {
+    // The unspecified state - most likey it is queued.
+    INSTANCE_STATUS_UNSPECIFIED = 0;
+    // Instance is starting. The first state update received from the
+    // instance.
+    STARTING = 1;
+    // The instance is running - one of two states of a healthy instance.
+    RUNNING = 2;
+    // The instance is idle - one of two states of a healthy instance.
+    IDLE = 3;
+    // The instance is stopping - usually follows from the RUNNING, IDLE,
+    // PREEMPT_REQUEST or DRAIN_REQUEST state.
+    STOPPING = 4;
+    // The instance is stopped - follows from the STOPPING state.
+    STOPPED = 5;
+    // The instance is in a bad state - but it is still able to send updates.
+    FAILING = 6;
+    // The subscribe service moves instances to this state if they
+    // have been idle for too long. This allows the cluster manager to
+    // make a final decision on whether or not to commence a drain
+    // sequence for this instance.
+    DRAIN_CONFIRMATION_PENDING = 7;
+    // An optional state that can be used to indicate that the instance
+    // is allocated from cloud provider, but ray hasn't been installed yet.
+    INSTANCE_ALLOCATED = 10;
+    // An optional state that can be used to indicate that the instance
+    // is currently installing Ray.
+    INSTALLING_RAY = 11;
+    // An optional state that can be used to indicate that the instance
+    // failed to allocate from cloud provider.
+    ALLOCATION_FAILED = 12;
+    // Node is deleted.
+    GARBAGE_COLLECTED = 13;
+  }
+  // an unique id for the instance that's generated by the
+  // instance manager. This may be optional if
+  // the instance hasn't be started yet.
+  string instance_id = 11;
+  // the status of the instance.
+  InstanceStatus status = 12;
```

Comment:
clarification: IIUC, if an instance is to be launched, it will be in `INSTANCE_STATUS_UNSPECIFIED` status?